modules/basegl/src/algorithm/dataminmaxgl.cpp

```
@@ -0,0 +1,436 @@
+        "u8vec4"sv,   // Vec4UInt8
+        "u16vec4"sv,  // Vec4UInt16
+        "uvec4"sv,    // Vec4UInt32
+        {},           // Vec4UInt64
+        // clang-format on
+    }};
+
+    if (glslType[static_cast<size_t>(id)].empty()) {
+        throw OpenGLException(IVW_CONTEXT_CUSTOM("GLFormat"),
+                              "Error no GLSL data type found for selected data format: {}",
+                              DataFormatBase::get(id)->getString());
+    }
+    return glslType[static_cast<size_t>(id)];
+}
+
+std::string_view glslDataConversion(size_t numComponents) {
+    // Conversion from any data type with numComponents channels to vec4.
+    // Special case for 3 component types.
+    // See glslDataType()
+    using namespace std::string_view_literals;
+    static constexpr std::array<std::string_view, 4> conversion{{
+        "vec4(value[index], 0, 0, 0)"sv,
+        "vec4(value[index], 0, 0)"sv,
+        "vec4(value[3 * index], value[3 * index + 1], value[3 * index + 2], 0)"sv,
+        "value[index]"sv,
+    }};
+    return conversion[std::min(numComponents - 1, conversion.size())];
+}
+
+}  // namespace detail
+
+DataMinMaxGL::DataMinMaxGL()
+    : gpuSupport_{isSuppportedByGPU()}
+    , volumeMinMax_{{{ShaderType::Compute, "compute/volumeminmax.comp"}}, Shader::Build::No}
+    , layerMinMax_{{{ShaderType::Compute, "compute/layerminmax.comp"}}, Shader::Build::No}
+    , linearMinMax_{{{ShaderType::Compute, "compute/linearminmax.comp"}}, Shader::Build::No}
+    , bufferMinMax_{{{ShaderType::Compute, "compute/bufferminmax.comp"}}, Shader::Build::No} {}
+
+std::pair<dvec4, dvec4> DataMinMaxGL::minMax(const Volume* volume) {
+    if (!gpuSupport_) {
+        return util::volumeMinMax(volume);
+    }
+    return minMax(volume->getRepresentation<VolumeGL>());
+}
+
+std::pair<dvec4, dvec4> DataMinMaxGL::minMax(const VolumeGL* volumeGL) {
+    if (!gpuSupport_) {
+        return util::volumeMinMax(volumeGL->getOwner());
```

Owner can be nullptr

---

Maybe it is better to throw an error here and have the Volume, overload handle the fallback... 
If you call it with a gl rep you are probably expecting it to use the gl rep also...